app/validators/linked_validator.rb

```diff
@@ -0,0 +1,44 @@
+class LinkedValidator < ActiveModel::EachValidator
+  DELIMITER = ':'.freeze
+
+  def validate_each(record, attribute, value)
+    register_linked = options[:register_linked]
+    record.errors.add attribute, 'Must be valid data from a register that is ready to use' unless validate_linked(value, register_linked)
+  end
+
+  def validate_linked(value, register_linked)
+    return false unless valid_value?(value)
+
+    if value.include? DELIMITER
+      check_curie(value)
+    else
+      check_register(value, register_linked)
+    end
+  end
+
+  def check_curie(value)
+    values = value.split(DELIMITER)
+
+    valid_key?(values[0], values[1])
+  end
+
+  def check_register(value, register_linked)
+    return false if register_linked.blank?
```

Comment:
could express this more concisely by e.g.
`register_linked.present? && valid_key?(register_linked, value)`
It will only run the second if the first is true.